app/views/timeline_events/show.html.erb

```diff
@@ -2,55 +2,105 @@
+    </div>
+    <div class="flex py-5 divide-x-2">
+      <div class="pr-4">
+        <p class="text-xs text-gray-500"><%= t(".submission_from") %></p>
+        <p class="text-sm font-semibold"><%= @submission.founder.name %></p>
+      </div>
+      <div class="px-4">
+        <p class="text-xs text-gray-500"><%= t(".submitted_on") %></p>
+        <p class="text-sm font-semibold"> <%= @submission.created_at.strftime("%b %d, %Y")%></p>
+      </div>
+      <div class="px-4">
+        <p class="text-xs text-gray-500"><%= t(".reviewed_on") %></p>
+        <p class="text-sm font-semibold"><%= @submission.evaluated_at.present? ? @submission.evaluated_at.strftime("%b %d, %Y") : t(".not_available") %></p>
+      </div>
+      <div class="px-4">
+        <p class="text-xs text-gray-500"><%= t(".status") %></p>
+        <% if @submission.passed_at.present? %>
+          <p class="text-sm font-semibold text-green-500"><%= t(".accepted") %></p>
+        <% elsif  @submission.evaluated_at.present? %>
+          <p class="text-sm font-semibold text-red-500"><%= t(".rejected") %></p>
+        <% else %>
+          <p class="text-sm font-semibold text-blue-500"><%= t(".pending") %></p>
         <% end %>
       </div>
-      <% @submission.checklist.each_with_index do |item, index| %>
-        <div class="mb-3 bg-gray-50 rounded">
-          <div class="text-lg font-bold ps-3 pt-3"><%= t('.question_with_number', number: index + 1) %></div>
-          <div class="convert-markdown p-3 pt-1 text-sm" data-json-props="<%= presenter.markdown_prop(item['title']) %>">
-            <%= item['title'] %>
-          </div>
-          <hr class="mt-3"/>
-          <% if presenter.render_markdown?(item) %>
-            <div class="convert-markdown p-3 text-sm" data-json-props="<%= presenter.markdown_prop(item['result']) %>"><%= item['result'] %></div>
-          <% elsif presenter.audio?(item) %>
-            <%= render 'audio', file: presenter.audio_file(item) %>
-          <% elsif presenter.files?(item) %>
-            <%= render 'files', files: presenter.files(item) %>
-          <% else %>
-            <div class="p-3 text-sm"><%= item['result'] %></div>
-          <% end %>
+    </div>
+  </div>
+</div>
+<div class="flex flex-col max-w-2xl mx-auto mt-10 bg-gray-50 p-4 rounded-md border border-gray-200 sm:px-2">
+  <% @submission.checklist.each_with_index do |item, index| %>
+    <div class="flex items-start gap-x-2">
+      <div class="mt-2">
+        <% if presenter.render_markdown?(item)%>
+          <i class="if i-long-text-light if-fw" ></i>
+        <% elsif presenter.link?(item) %>
+          <i class="if i-link-light if-fw" ></i>
+        <% elsif presenter.audio?(item) %>
+          <i class="if i-file-music-light if-fw" ></i>
+        <% elsif presenter.files?(item)%>
+          <i class="if i-file-light if-fw" ></i>
+        <% else %>
+          <i class="if i-check-circle-alt-regular if-fw" ></i>
+        <% end %>
```

Comment:
Why not switch on the `kind` 🤔

Reply:
I think, using case-when will introduce string literals while comparing.
Should I use case-when?

Reply:
Yes, using a case-when statement might introduce string literals, but it can help make your code cleaner and more readable. Plus, it is more idiomatic in Ruby.

Here's how you can refactor your code using case-when:

```rb
def icon_class_for(item)
  case item['kind']
  when 'longText', 'shortText', 'link'
    'if i-long-text-light if-fw'
  when 'files'
    'if i-file-light if-fw'
  when 'audio'
    'if i-file-music-light if-fw'
  else
    'if i-check-circle-alt-regular if-fw'
  end
end
```
Now you can just use this method in your view:

```rb
<i class="<%= presenter.icon_class_for(item) %>"></i>
```
This not only reduces the complexity of the view, but it also moves the responsibility of determining the icon to the presenter, which is where this logic should live.

As for your concern about string comparisons, I wouldn't worry too much about it. The performance difference is negligible for a small number of strings, and the improved readability and structure is worth it.

Reply:
This looks good! 😄

Reply:
Thanks @bodhish  for the explanation!